amqtt/contexts.py

```diff
@@ -1,22 +1,357 @@
-from enum import Enum
+from dataclasses import dataclass, field, fields, replace
```

Comment:
I really like the direction this is going! <3

For parsing I really do prefer pydantic as it allows more lenient parsing as well as generating docs from the definitions via JSON Schema.

It would add a somewhat big dependency.

Reply:
i like pydantic. but decided against it due to its dependency footprint.